docs/proposals/vision.md

```diff
@@ -0,0 +1,130 @@
+than on the super-fresh metrics stored in the memory.
+UC5 can go either with UC1/UC2 or with UC4. As there is no immediate need for UC4 we will
+ provide basic UC5 together with UC1/UC2 but in the future it will join UC4.
+
+This separation leads to an idea of splitting Heapster into 3 binaries:
+
+* Core Heapster - covering UC1, UC2 and temporarily UC5
+* Eventer - covering UC3
+* Oldtimer - covering UC4 and UC5 
+
+## Reduction of Responsibility
+
+With 3 possible node sources (Kuberentes API Server, flat file, CoreOS Fleet), 2 metrics 
+sources (cAdvisor and Kubelet) and constantly growing number of sinks we have to separate 
+the stuff that the core Heapster/K8S team is responsible for and what is provided as a 
+plugin/addition and doesn’t come in the main release package. 
+
+We decided to focus only on:
+
+Kubernetes API Server node source
+Kubelet metrics source
+Influxdb, GCM, GKE (there is special endpoint for GKE that exposes all available metrics),
+ Hawkular sinks for Heapster
+Influxdb, GCL sinks for Eventer
+
+The rest of the sources/sinks will be available as plugins. The plugin will be used in 2 flavors:
+
+* Complied in - will require the user to rebuild the package and create his own image with 
+the desired set of plugins.  
+* Side-car - Heapster will talk to plugin’s HTTP server to get/pass metrics through a well 
+defined json interface. The plugin runs in a separate container.
+
+K8s team will explicitly say that it is NOT giving any warranty on the plugins. Plugins e2e 
+tests can be included in some CI suite but we will not block our development (too much) if 
+something breaks Kafka or Riemann. We will also not pay attention to whether a particular sink scales up.
+
+For now we will keep all of the currently available sinks compiled-in by default, to keep the 
+new Heapster more or less compatible with the old one, but eventually (if the number of sinks grows)
+ we will migrate some of them to plugins.
+
+## Custom Metrics Status
+
+Heapster is not a generic solution for gathering arbitrary number of arbitrary-formated custom 
+metrics. The support for custom metrics is focused on auto-scaling and critical functionality 
+monitoring (and potentially scheduling). And Heapster is oriented towards system metrics, not 
+application/business level metrics.
+
+Kubernetes users and application developers will be able to push any number of their custom 
+metrics through our pipeline to the storage but this should be considered as a bonus/best effort 
+functionality. Custom metrics will not influence our performance targets (no extra fine-tunning effort 
+to support >5 custom metrics per pod). There will be a flag in Kubelet that will limit the 
+number of custom metrics.
+
+## Performance Target
+
+Heapster product family (Core, Eventer and Oldtimer) should follow the same performance goals 
+as core Kubernetes. As Eventer is fairly simple and Oldtimer not yet fully defined this section
```

Comment:
There is no official Heapster/Metrics api yet. There is some api for stats but I'm not sure if will be exactly reproduced in Oldtimer.